src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/AbstractExpressionHandler.java

```diff
@@ -303,22 +303,21 @@ private void checkLinesIndent(LineSet lines,
     /**
      * Check the indentation for a single line.
      *
-     * @param lineNum       the number of the line to check
-     * @param colNum        the column number we are starting at
+     * @param ast           the abstract syntax tree to check
      * @param indentLevel   the indentation level
      * @param mustMatch     whether or not the indentation level must match
      */
-    private void checkLineIndent(int lineNum, int colNum,
+    private void checkLineIndent(DetailAST ast,
         IndentLevel indentLevel, boolean mustMatch) {
-        final String line = indentCheck.getLine(lineNum - 1);
+        final String line = indentCheck.getLine(ast.getLineNo() - 1);
         final int start = getLineStart(line);
         // if must match is set, it is a violation if the line start is not
         // at the correct indention level; otherwise, it is an only an
         // violation if this statement starts the line and it is less than
         // the correct indentation level
         if (mustMatch && !indentLevel.isAcceptable(start)
-                || !mustMatch && colNum == start && indentLevel.isGreaterThan(start)) {
-            logChildError(lineNum, start, indentLevel);
+                || !mustMatch && ast.getColumnNo() == start && indentLevel.isGreaterThan(start)) {
```

Comment:
>  When I update indentation to log DetailAST, I run com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java to test, but one problem appear, this test failure:

There were some bugs in the code due to which it was not taking tabspaces, now I have fixed it in *PR* #8038.

# Reason for Failure:
```
 final int columnNumber = expandedTabsColumnNo(ast);
        // if must match is set, it is a violation if the line start is not
        // at the correct indention level; otherwise, it is an only an
        // violation if this statement starts the line and it is less than
        // the correct indentation level
        if (mustMatch && !indentLevel.isAcceptable(start)
                || !mustMatch && colNum == start && indentLevel.isGreaterThan(start)) {
            logChildError(lineNum, start, indentLevel);
                || !mustMatch && columnNumber == start && indentLevel.isGreaterThan(start)) {
            logChildError(ast, start, indentLevel);
```
ast.getColumnNumber is creating the problem because columnNumber() dont unfold the tabspaces so wrong linestart log is created by it.

Reply:
Thanks @Abhishek-kumar09 , it is welcome if you are willing to continue to finish your PR on indentation. This PR is based on your PR and the reason that push this PR is that the original PR has not been replied for a long time

Reply:
@HuGanghui I am working on it, Indentation is one of the most complex fields in Checkstyle. The changes is not only limited to one or two checks but there are many subchecks inside it,that create complexity. 
Well you don't have to worry I will be fixing all those myself. You can untill focus on your next goals. Goodluck.

Reply:
Ok, lets @Abhishek-kumar09  to finish his PR ...
@HuGanghui , please keep your draft for now, we might come back to it if we loss connection to @Abhishek-kumar09 .

Reply:
OK, got it, thanks @Abhishek-kumar09